github/examples_test.go

```diff
@@ -74,3 +74,24 @@ func ExampleUsersService_ListAll() {
 		// Process users...
 	}
 }
+
+func ExamplePullRequestsService_Create() {
+	// In this example we're creating a PR and displaying the HTML url at the end.
+	client := github.NewClient(nil)
```

Comment:
Not dealing with authentication keeps the example shorter and simpler, but I worry it might be misleading.

In reality, this code would never work as is, because creating a PR requires an authenticated user.

I think it'd be good if we added a comment noting that, and perhaps linking to https://godoc.org/github.com/google/go-github/github#hdr-Authentication for more information on how to do authentication.

What do you think?